### PR TITLE
Updates to control flow support in dependency analysis

### PR DESCRIPTION
**Pull request by atgeller**
### Description

Refactors the analysis to work in the expected order, so now allocation and lifting work as expected.

### CODE REVIEW

File: test/Quake/dependency-1.qke [hunk 1]
```diff
@@ -6,11 +6,11 @@
 // the terms of the Apache License 2.0 which accompanies this distribution.   //
 // ========================================================================== //
 
-// RUN: cudaq-opt --assign-ids --dep-analysis %s | FileCheck %s
+// RUN: cudaq-opt --add-wireset --assign-wire-indices %s | FileCheck %s
```
Comment (bmhowe23) [inline test/Quake/dependency-1.qke:9]: Are these tests passing for you? It looks like they fail for me:
```
Failed Tests (6):
  CUDAQ :: Quake/dependency-1.qke
  CUDAQ :: Quake/dependency-2.qke
  CUDAQ :: Quake/dependency-3.qke
  CUDAQ :: Quake/dependency-4.qke
  CUDAQ :: Quake/dependency-classical.qke
```

**Review by bmhowe23 — COMMENTED** [review]
Thanks, Adam. I'll download this PR to start playing with it, but would you mind updating it to address the compilation errors when using the CI's compiler: https://github.com/NVIDIA/cuda-quantum/actions/runs/10490052305/job/29060785461#step:7:921

**Review by bmhowe23 — APPROVED** [review]
I'm ok pulling this change in now and working the comments afterwards. I also think we need a fresh merge from main after incorporating this change into the feature branch.

File: test/Quake/dependency-1.qke (hunk 1 above)
Comment (schweitzpgi) [reply test/Quake/dependency-1.qke:9]: Those failures look familiar. I thought there was a patch to fix them, but maybe I'm recalling the wrong thing?

**Pull request merged**